Updated upstream Bitnami image versions

agronholm opened the pull request:
The previous images did not provide the correct POSTGRESQL_DATA_DIR environment variable, causing the Timescale init scripts to fail.

Fixes #50, fixes #56.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/timescale/timescaledb-docker?pullRequest=59) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/timescale/timescaledb-docker?pullRequest=59) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/timescale/timescaledb-docker?pullRequest=59) it.</sub>

agronholm commented:
It looks like this requires some more tuning after all.

RobAtticus commented:
The changes look good, but what do you mean by more tuning?

agronholm commented:
The upstream images contain other logic changes which cause the container to not start properly. For example, it now requires that `POSTGRES_PASSWORD` is defined. Also, it sets a switch that causes unbound variable access in shell scripts to fail, terminating the entire script. This is a problem when the init scripts check for `POSTGRES_USER`on line 5 in `000_install_timescaledb.sh`.

agronholm commented:
[This](https://github.com/bitnami/bitnami-docker-postgresql/blob/11.2.0-debian-9-r74/11/debian-9/rootfs/entrypoint.sh#L4) is the offending line in the upstream entry point.

agronholm commented:
Now the scripts no longer crash, but apparently the timescaledb extension does not get loaded at the start. Investigating.

agronholm commented:
This happens because there is absolutely nothing in these images or init scripts that would add the necessary `shared_preload_libraries='timescaledb'` to the PostgreSQL configuration.

agronholm commented:
Now it finally works.

RobAtticus commented on bitnami/Dockerfile:
```diff
@@ -153,5 +153,6 @@ RUN set -ex \
         /var/lib/apt/lists/* \
         /tmp/*               \
         /var/tmp/*
+RUN sed -r -i "s/[#]*\s*(shared_preload_libraries)\s*=\s*'(.*)'/\1 = 'timescaledb,\2'/;s/,'/'/" /opt/bitnami/postgresql/share/postgresql.conf.sample
```
This should be unnecessary since `timescaledb-tune` will add it to the shared_preload_libraries -- can you see if there is a reason that isn't happening?

agronholm replied:
I'll look into it and modify the PR if I find something actionable.

agronholm replied:
The startup process crashes on something before it has a chance to run `timescaledb-tune`.

agronholm replied:
Right. The first init script already tries to do `CREATE EXTENSION timescaledb` but that fails because the library hasn't been preloaded.

agronholm replied:
So it works if I either:

1. Modify `postgresql.conf.sample` in the Dockerfile (like the main image does)
1. Skip the creation of the timescale extension in the `template` and `postgres` databases

RobAtticus replied:
Ah, good catch then. Thank you!

agronholm commented:
That's it for my tuning. I would also recommend to do the compilation step before copying the scripts because otherwise you'll have to recompile everything for any little change you do to the scripts, but that is beyond the scope of this PR.

agronholm commented:
> Looks good, would you mind squashing the commits and updating the commit msg to summarize all the changes in one?

Do you not have squash merges enabled for this repository? Github added this feature quite a while ago.

I'm also looking at a cleaner alternative to editing the sample config file. I realize that simply adding the preload line as a separate file to `conf.d` might work. Would you prefer this if it works?

agronholm commented:
I now squashed my changes. I also went ahead and changed the way the extension is preloaded by adding the `conf.d/timescaledb.conf` file instead of modifying the sample config file. I can revert this change if you don't like it.

RobAtticus commented:
I think the previous way was actually more robust in case there are ever other shared_preload_libraries.

agronholm commented:
Alright, I've reverted the change now.

RobAtticus commented on bitnami/Dockerfile:
```diff
@@ -153,5 +153,6 @@ RUN set -ex \
         /var/lib/apt/lists/* \
         /tmp/*               \
         /var/tmp/*
+RUN echo "shared_preload_libraries = 'timescaledb'" >/opt/bitnami/postgresql/conf/conf.d/timescaledb.conf
```
I still see the newer fix here instead of the `sed` one?

agronholm replied:
Sorry, seems I forgot to commit the change. It's fixed now.

RobAtticus commented:
Excellent, much appreciated!

agronholm commented:
I discovered another problem. Replication does not work. The slave container fails to start because the tuning script changes the master's `max_worker_processes` value. When the slave tries to initialize, it sees this discrepancy and bails before the tuning script has even had a chance to run. How did you get replication to work before?

agronholm commented:
Setting `TS_NO_TUN` skips the tuning but another problem follows: the container attempts to run the scripts on the slave. I don't think it should, but even in the upstream image I cannot see anything that would prevent this. Thoughts?
